WalletWasabi/WabiSabi/Models/DecompositionAlgs/Decomposition.cs

```diff
@@ -0,0 +1,59 @@
+using NBitcoin;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public class Decomposition : IEnumerable<Coin>
+	{
+		private List<Coin> Coins { get; } = new();
+
+		public void Extend(Coin coin)
+		{
+			int index = Coins.BinarySearch(coin);
+			if (index < 0)
+			{
+				Coins.Insert(~index, coin);
+			}
+		}
+
+		public Money FaceValue()
+		{
+			return Coins.Sum(c => c.Amount);
+		}
+
+		public int OutputsVbytes()
+		{
+			return Coins.Sum(_ => Coin.OutputVbytes);
+		}
+
+		public decimal InputsVbytes()
+		{
+			return Coins.Sum(_ => Coin.InputVbytes);
+		}
+
+		public Money EffectiveValue(FeeRate feeRate)
+		{
+			var sats = (long)Math.Floor(feeRate.SatoshiPerByte * Coin.InputVbytes * Coins.Count);
+			return FaceValue() - Money.Satoshis(sats);
```

Comment:
```suggestion
			return FaceValue() - (feeRate.GetFee(Coin.InputVbytes) * Coins.Count);
```

IIRC you can multiply money by int, but even if not `GetFee` should be used

Reply:
ACK